File: py/src/ticket_tracker/paths.py
"""XDG 路径解析 + macOS 兼容兜底。"""

import os
from pathlib import Path

APP_NAME = "ticket-tracker"


def _xdg(env_var, default):
    val = os.environ.get(env_var)
    return Path(val).expanduser() if val else Path(default).expanduser()


def config_dir():
    """XDG_CONFIG_HOME 或 macOS 兜底 ~/.config/<APP_NAME>。"""
    base = _xdg("XDG_CONFIG_HOME", "~/.config")
    p = base / APP_NAME
    p.mkdir(parents=True, exist_ok=True)
    return p


def state_dir():
    """运行期状态/日志位置。XDG_STATE_HOME 或 macOS 兜底 ~/.local/state。"""
    base = _xdg("XDG_STATE_HOME", "~/.local/state")
    p = base / APP_NAME
    p.mkdir(parents=True, exist_ok=True)
    return p


def config_file():
    return config_dir() / "config.json"


def log_file():
    return state_dir() / "ticket-tracker.log"


def pid_file():
    return state_dir() / "ticket-tracker.pid"

File: rs/src/paths.rs
//! XDG 路径解析 —— 与 py/.../paths.py 1:1 对齐（见 RUST_PORT.md §4.4）。

use std::path::{Path, PathBuf};

const APP_NAME: &str = "ticket-tracker";

/// `XDG_CONFIG_HOME` 或 `~/.config`，叠加 `ticket-tracker/`。
///
/// 注意：故意不使用 `dirs::config_dir()`（macOS 会回到 `~/Library/Application Support`），
/// 强迫走 XDG 路径，与 Python 版 1:1 兼容（已存在的配置在 `~/.config/ticket-tracker/`）。
pub fn config_dir() -> PathBuf {
    let base = std::env::var("XDG_CONFIG_HOME")
        .ok()
        .filter(|s| !s.is_empty())
        .map(PathBuf::from)
        .unwrap_or_else(|| PathBuf::from("~/.config"));
    let p = expand(base).join(APP_NAME);
    ensure_dir(&p);
    p
}

/// `XDG_STATE_HOME` 或 `~/.local/state`，叠加 `ticket-tracker/`。
pub fn state_dir() -> PathBuf {
    let base = std::env::var("XDG_STATE_HOME")
        .ok()
        .filter(|s| !s.is_empty())
        .map(PathBuf::from)
        .unwrap_or_else(|| PathBuf::from("~/.local/state"));
    let p = expand(base).join(APP_NAME);
    ensure_dir(&p);
    p
}

fn expand(p: PathBuf) -> PathBuf {
    let s = p.to_string_lossy();
    if s.starts_with("~/") {
        if let Some(home) = dirs::home_dir() {
            return home.join(s.trim_start_matches("~/"));
        }
    }
    p
}

pub fn config_file() -> PathBuf {
    config_dir().join("config.json")
}

pub fn log_file() -> PathBuf {
    state_dir().join("ticket-tracker.log")
}

pub fn pid_file() -> PathBuf {
    state_dir().join("ticket-tracker.pid")
}

fn ensure_dir(p: &Path) {
    let _ = std::fs::create_dir_all(p);
}
